### PR TITLE
fix audio network stats

### PR DESCRIPTION
**Pull request by zzmp**

### CODE REVIEW

**Comment by zzmp** [comment]
QA:

- The audio stats should still work: `Developer > Audio > Audio Network Stats...`; they should now be more intelligible
- On this build (DEV and PR builds only) there should be additional audiostream logging, telling you how the jitter buffer is behaving
- Smoke test audio

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9792d025fa1186ab80e9ebcd871b17c088b5ed9a available at: https://gist.github.com/7b22b313f2174527dc71d1c3f01198a9<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8609-9792d02.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8609-9792d02.dmg

**Comment by zzmp** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9792d025fa1186ab80e9ebcd871b17c088b5ed9a available at: https://gist.github.com/7b22b313f2174527dc71d1c3f01198a9<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8609-9792d02.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8609-9792d02.dmg

File: interface/src/ui/AudioStatsDialog.cpp [hunk 1]
```diff
@@ -110,137 +110,135 @@ void AudioStatsDialog::renderStats() {
...
     clearAllChannels();
     
+    double mixerRingBufferFrames = 0.0,
+        outputRingBufferFrames = 0.0;
     double audioInputBufferLatency = 0.0,
-           inputRingBufferLatency = 0.0,
-           networkRoundtripLatency = 0.0,
-           mixerRingBufferLatency = 0.0,
-           outputRingBufferLatency = 0.0,
-           audioOutputBufferLatency = 0.0;
```
Comment (Atlante45) [inline interface/src/ui/AudioStatsDialog.cpp]: Not a big fan of that formatting. I actually thought it was disallowed in the coding standard, but I couldn't find the ref, so maybe not.
At least make it in just one block in the case you want to keep it that way so it is consistent.

File: interface/src/ui/AudioStatsDialog.cpp [hunk 2]
```diff
@@ -82,7 +82,7 @@ AudioStatsDialog::AudioStatsDialog(QWidget* parent) :
     _upstreamInjectedID = addChannel(_form, _upstreamInjectedStats, COLOR0);
         
     connect(averageUpdateTimer, SIGNAL(timeout()), this, SLOT(updateTimerTimeout()));
-    averageUpdateTimer->start(1000);
+    averageUpdateTimer->start(200);
```
Comment (Atlante45) [inline interface/src/ui/AudioStatsDialog.cpp]: I've noticed in the past, that the audio stats window shows up and stays empty for a second or so.
That was probably why, you should call `updateTimerTimeout()` once here too to pre-populate the window.

Comment (zzmp) [inline interface/src/ui/AudioStatsDialog.cpp]: This did cause the text to appear, but the window still starts out incorrectly sized. I tried a call to `adjustSize` and it did not fix it. I don't know how to fix it :(. Hopefully it's not bad enough to hold up the PR, it is a debug tool.

**Comment by hifi-gustavo** [comment]
Build console output for commit: bbbe070af10f2dd8262876be0544fc5c22e4db7f available at: https://gist.github.com/7b22b313f2174527dc71d1c3f01198a9<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8609-bbbe070.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8609-bbbe070.dmg

**Comment by zzmp** [comment]
This crashes when trying to echo local audio. DNM until I can investigate tomorrow.

@Atlante45 feel free to code review in the meantime.

**Comment by zzmp** [comment]
[It was crashing because my audio formats were different for my speakers and microphone](https://highfidelity.fogbugz.com/f/cases/1699/Bugsplat-5746-If-Mic-and-Speakers-have-different-sample-rates-audio-formats-Echo-local-audio-will-crash).

File: libraries/audio-client/src/AudioClient.cpp [hunk 3]
```diff
@@ -941,6 +941,10 @@ void AudioClient::handleAudioInput() {
 
         emitAudioPacket(encodedBuffer.constData(), encodedBuffer.size(), _outgoingAvatarAudioSequenceNumber, audioTransform, packetType, _selectedCodecName);
         _stats.sentPacket();
+
+        int bytesInInputRingBuffer = _inputRingBuffer.samplesAvailable() * sizeof(int16_t);
```
Comment (Atlante45) [inline libraries/audio-client/src/AudioClient.cpp]: Pretty sure there's a `SAMPLE_SIZE` constant.

Comment (zzmp) [inline libraries/audio-client/src/AudioClient.cpp]: There's not. There is a typedef of int16_t to AudioConstants::AudioSample.

**Comment by hifi-gustavo** [comment]
Build console output for commit: b6d093889c2e3baa1562f61242cdd9d1dba1c44d available at: https://gist.github.com/7b22b313f2174527dc71d1c3f01198a9<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8609-b6d0938.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8609-b6d0938.dmg

File: libraries/audio-client/src/AudioClient.cpp (hunk 3 above)
Comment (kencooke) [inline libraries/audio-client/src/AudioClient.cpp]: Samples can be int16_t or float, depending on the context, so hiding it behind a constant might be bug-prone

Comment (zzmp) [inline libraries/audio-client/src/AudioClient.cpp]: We already put `AudioSample` behind a constant, so I think it is safe to assume the developer should know when they are not using the constants (i.e. using float samples) even with this change.

**Comment by OmegaHeron** [comment]
Testing Windows 10
Stats display proper and, yes, make sense and offer > clarity over form implementation.  All general smoke tests pass.  I didn't get my oscilloscope out and measure latency.  :)  I'd use a different tool for that if I did so, but that's a job for the HF internals.  QA Pass

**Pull request merged**